app/src/main/java/org/xbmc/kore/service/library/SyncMusic.java

```diff
@@ -347,12 +348,28 @@ public void onSuccess(ApiList<AudioType.DetailsSong> result) {
 
                 // Save partial results to DB
                 ContentValues songValuesBatch[] = new ContentValues[items.size()];
+                int artistsCount = 0;
                 for (int i = 0; i < items.size(); i++) {
                     AudioType.DetailsSong song = items.get(i);
                     songValuesBatch[i] = SyncUtils.contentValuesFromSong(hostId, song);
+
+                    artistsCount += song.artistid.size();
                 }
                 contentResolver.bulkInsert(MediaContract.Songs.CONTENT_URI, songValuesBatch);
 
+                // Iterate on each song, collect the artists and insert them
+                ContentValues songArtistsValuesBatch[] = new ContentValues[artistsCount];
+                int artistCount = 0;
+                for (AudioType.DetailsSong song : items) {
+                    for (int artistId : song.artistid) {
+                        songArtistsValuesBatch[artistCount] = new ContentValues();
+                        songArtistsValuesBatch[artistCount].put(MediaContract.SongArtists.HOST_ID, hostId);
+                        songArtistsValuesBatch[artistCount].put(MediaContract.SongArtists.SONGID, song.songid);
+                        songArtistsValuesBatch[artistCount].put(MediaContract.SongArtists.ARTISTID, artistId);
+                        artistCount++;
+                    }
+                }
+
```

Comment:
You're right. I updated the code and added the bulkInsert. Sorry about the mess. 

(I don't have a lot of time at the moment. After your first comment I quickly compared this with #251. I overlooked the code refactoring in #251 which adds insertSongs so it can be used from the tests as well. This made me think the above code was a leftover from the previous solution)